### DISCUSSION TITLE
Expose `pagination` state through API

### DISCUSSION THREAD

**Discussion by dchambers** [TanStack/table#5551] [Ideas] [answered]
Since there is no `table.pagination` API there is currently no way to implement working pagination support except by owning the pagination state externally. This is different to many other features (e.g. column sorting & row selection) where you create a workable UI using the provided APIs only.

**Comment by KevinVandy** [comment, marked as answer]
I don't understand this question at all. There are plenty of ways to access pagination state.

You can access any state in the table at all times with the `table.getState()` API: https://tanstack.com/table/latest/docs/framework/react/guide/table-state#accessing-table-state

<img width="956" alt="image" src="https://github.com/TanStack/table/assets/28243511/9acd1b46-843c-4293-ab11-3d5e293655c5">

Here's the whole paginition guide too: https://tanstack.com/table/latest/docs/guide/pagination

**Reply by dchambers** [reply]
> You can access any state in the table at all times with the table.getState() API:

Awesome, it works! I didn't realise this table-level escape hatch existed. Thanks for you saving me from continuing to write lots of gnarly work around code 🙏
